app/src/main/java/com/example/trainingintive/repository/DogImageRepository.kt

```diff
@@ -1,11 +1,31 @@
 package com.example.trainingintive.repository
 
 import com.example.trainingintive.model.DogImageUrl
+import com.example.trainingintive.repository.local.DogImageDao
 import com.example.trainingintive.repository.network.DogImageApiService
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.core.Flowable
 import io.reactivex.rxjava3.core.Single
+import io.reactivex.rxjava3.disposables.Disposable
+import io.reactivex.rxjava3.schedulers.Schedulers
 import javax.inject.Inject
 
-class DogImageRepository @Inject constructor(private val dogImageApiService: DogImageApiService) {
+class DogImageRepository @Inject constructor(
+    private val dogImageApiService: DogImageApiService,
+    private val dogImageDao: DogImageDao
+) {
 
     fun getDogImageUrl(): Single<DogImageUrl> = dogImageApiService.getDogImage()
+
+    fun getAllImageUrlsFromLocalDatabase(): Flowable<List<DogImageUrl>> = dogImageDao.getAllImageUrls()
+
+    fun insertIntoLocalDatabase(dogImageUrl: DogImageUrl): Disposable {
+        return Single.fromCallable({ dogImageDao.insert(dogImageUrl) })
+            .subscribeOn(Schedulers.io())
```

Comment:
No! No! You should subscribe in ViewModel not in Repository. Look at my other comment (about using Completable), repo should return completable to ViewModel, and you should subscribe in ViewModel.